actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -208,6 +208,55 @@ class Cookies
     # Raised when storing more than 4K of session data.
     CookieOverflow = Class.new StandardError
 
+    # Parse the set-cookie header and return a hash of cookie names and values.
+    #
+    # Example:
+    #   set_cookies = headers["set-cookie"]
+    #   parse_set_cookies_headers(set_cookies)
+    def self.parse_set_cookies_headers(set_cookies)
```

Comment:
Why we need to parse cookies now?

Reply:
A ton of tests depend on it and they were just using string comparisons which is insufficient. I don't know where else to put this, can you suggest a better place? Somewhere in the test helpers?

Reply:
I've moved this code into the test helpers, hopefully that's a better place for it.